src/org/mozilla/javascript/NativeObject.java

```diff
@@ -567,6 +567,12 @@ public Object execIdCall(IdFunctionObject f, Context cx, Scriptable scope,
               Scriptable sourceObj = ScriptRuntime.toObject(cx, thisObj, args[i]);
               Object[] ids = sourceObj.getIds();
               for (Object key : ids) {
+                if (targetObj instanceof ScriptableObject) {
+                  ScriptableObject desc = ((ScriptableObject) targetObj).getOwnPropertyDescriptor(cx, key);
+                  if (desc != null && isDataDescriptor(desc) && isFalse(desc.get("writable"))) {
+                      throw ScriptRuntime.typeError1("msg.change.value.with.writable.false", key);
```

Comment:
Thanks! This looks great except that since I just merged your other PR this is now deprecated. Can you please update it?

Reply:
I changed it to use `ScriptRuntime.typeErrorById` and squashed it.